src/helpers/simplify_claims.ts

```diff
@@ -1,176 +1,166 @@
   }
 
   if (keepIds) valueObj.id = claim.id
 
   return valueObj
 }
 
-export function simplifyClaims (claims: Claims, options: SimplifySnaksOptions = {}): SimplifiedClaims {
-  return simplifySnaks(claims, options)
+export function simplifyClaims (claims: Claims, options: SimplifyClaimsOptions = {}): SimplifiedClaims {
+  return applyObjectSimplification(claims, simplifyPropertyClaims, options)
 }
-export function simplifyPropertyClaims (propertyClaims: PropertyClaims, options: SimplifySnaksOptions = {}): SimplifiedPropertyClaims {
-  return simplifyPropertySnaks(propertyClaims, options)
+
+export function simplifyPropertyClaims (propertyClaims: PropertyClaims, options: SimplifyClaimsOptions = {}): SimplifiedPropertyClaims {
+  // Avoid to throw on empty inputs to allow to simplify claims array
+  // without having to know if the entity as claims for this property
+  // Ex: simplifyPropertyClaims(entity.claims.P124211616)
+  if (propertyClaims == null || propertyClaims.length === 0) return []
+
+  const { keepNonTruthy, keepNonDeprecated } = options
+
+  if (keepNonDeprecated) {
+    propertyClaims = nonDeprecatedPropertyClaims(propertyClaims)
+  } else if (!(keepNonTruthy)) {
+    propertyClaims = truthyPropertyClaims(propertyClaims)
+  }
+
+  return applyArraySimplification(propertyClaims, simplifyClaim, options)
 }
-export function simplifyClaim (claim: Claim, options: SimplifySnakOptions = {}): SimplifiedClaim {
-  return simplifySnak(claim, options)
+
+export function simplifySnaks (snaks: Snaks = {}, options: SimplifySnaksOptions = {}): SimplifiedSnaks {
+  return applyObjectSimplification(snaks, simplifyPropertySnaks, options)
+}
+
+export function simplifyPropertySnaks (propertySnaks: PropertySnaks, options: SimplifySnaksOptions = {}): SimplifiedPropertySnaks {
+  if (propertySnaks == null || propertySnaks.length === 0) return []
+  return applyArraySimplification(propertySnaks, simplifySnak, options)
+}
+
+function applyObjectSimplification (obj, fn, options) {
+  const { propertyPrefix } = options
+  const simplified = {}
+  for (let [ propertyId, propertyArray ] of Object.entries(obj)) {
+    if (propertyPrefix) {
+      propertyId = propertyPrefix + ':' + propertyId
+    }
+    simplified[propertyId] = fn(propertyArray, options)
+  }
+  return simplified
+}
+
+function applyArraySimplification (array, simplifyFn, options) {
+  const simplifiedArray = array
+    .map(claimOrSnak => simplifyFn(claimOrSnak, options))
+    // Filter-out novalue and somevalue claims,
+    // unless a novalueValue or a somevalueValue is passed in options
+    // Considers null as defined
+    .filter(obj => obj !== undefined)
+
+  // Deduplicate values unless we return a rich value object
+  if (simplifiedArray[0] && typeof simplifiedArray[0] !== 'object') {
+    return uniq(simplifiedArray)
+  } else {
+    return simplifiedArray
+  }
 }
 
 export function simplifyQualifiers (qualifiers: Qualifiers, options: SimplifySnaksOptions = {}) {
```

Comment:
accept unknown as an argument (typescript doesn't notice this as long as `apply*Simplification` isn't typed)

```suggestion
export function simplifyQualifiers (qualifiers: Qualifiers = {}, options: SimplifySnaksOptions = {}) {
```

Reply:
I added types to `apply*Simplification` functions in c0a5f8a, would that `= {}` still be needed now?